### DISCUSSION TITLE
Themed strips

### DISCUSSION THREAD

**Discussion by jdicko** [teunbrand/ggh4x#115] [General]
Hello,

I was wondering about the possibilities  with setting the strip backgrounds in a nested facet. I want to set the strip background colours of upper strip whilst keeping the lower strip as is. e.g i  have facet_nested_wrap(~ _i_ + _j_), i just want to colour the strip background for _i_ (and conditionally if possible based upon values of i)

Any feedback would be appreciated

**Comment by teunbrand** [comment]
You can't really 'map' a colour to the strip background, you'd have to provide them manually. In `elem_list_rect()`, `NA` substitutes the default instead of meaning 'transparent', so you can use that to keep the lower strips as per the theme settings.

``` r
library(ggh4x)
#> Loading required package: ggplot2

# There are 5 upper nested strips and 7 individual strips beneath
colours <- c("dodgerblue", "limegreen", "tomato", "orchid", "gold",
             rep(NA, 7))

ggplot(mpg, aes(hwy, cty)) +
  geom_point() +
  facet_nested_wrap(
    vars(cyl, year),
    strip = strip_nested(
      background_x = elem_list_rect(fill = colours)
    )
  )
```

![](https://i.imgur.com/VyxgjFn.png)

<sup>Created on 2023-07-18 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by jdicko** [comment]
Thanks, this is exactly what I was after, much appreciated
